utils/term_analysis.cpp

```diff
@@ -50,4 +52,58 @@ UnorderedTermSet get_free_symbols(const Term & term)
   get_free_symbols(term, free_symbols);
   return free_symbols;
 }
+
+void get_predicates(const Term & term,
+                    const Sort & boolsort,
+                    UnorderedTermSet & out,
+                    bool include_symbols)
+{
+  TermVec to_visit({ term });
+  UnorderedTermSet visited;
+
+  // set of boolean operators
+  // boolean terms with these operators are not predicates
+  unordered_set<PrimOp> boolops({ And, Or, Xor, Not, Implies, Iff, Ite });
```

Comment:
I am not sure how do we are extracting predicates in the case of Ite?
It should be different from the other cases, right?

Reply:
I think it's the same. In this case, `boolops` are the operators that are ignored and so it just recurses on all the children of that term. Although there might be a slight difference depending on what we count as predicates, see comment below.

Reply:
We discussed offline that the ITEs should be handled specially for performance reasons, i.e. for `y = ite(x < 10, x, 0)` we should find predicates `x < 10`, `y = x` and `y = 0` instead of using the whole equality with an ite. Made that change with 612bc97ee8f414511b19af46fdac7436cfe9ce55.